### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by kostiantin-hudzenko**
[DEMO LINK](https://kostiantin-hudzenko.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +16,101 @@ export const goodsFromServer = [
...
-          Reset
-        </button>
+        {(sortField !== '' || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setIsReversed(false);
+              setSortField('');
+            }}
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:103]: make handler in component and pass reference here
```
const handleReset = () => {
  setIsReversed(false);
  setSortField('');
}
```
```suggestion
            onClick={handleReset}
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +16,101 @@ export const goodsFromServer = [
...
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField !== '' || isReversed) && (
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:96]: better to make separate enum field for default sort.

```
enum SortType {
  none = 'none',
  alphabet = 'alphabet',
  length = 'length',
}
```

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +16,101 @@ export const goodsFromServer = [
...
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.alphabet,
+          })}
+          onClick={() => setSortField(SortType.alphabet)}
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:71]: for both sort buttons make one handler factory which will accept SortType and return arrow function
```
const hanldeSort = (sort: SortType) => () => setSortField(sort);
```
```suggestion
          onClick={handleSort(SortType.alphabet)}
```

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +16,101 @@ export const goodsFromServer = [
...
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+          onClick={() => setSortField(SortType.length)}
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:81]: same as for reset make handler and pass reference here.
it is bad practise to make anonymous arrow functions in jsx.

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Good job.
Please check a few minor comments.

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
You should check demo limk before request review
2023-10-02_15h32_51

**Pull request closed without merging**